charm/kafka/reactive/tls.py

```diff
@@ -163,3 +139,86 @@ def import_ca_crt_to_keystore():
 
             remove_state('tls_client.ca_installed')
             set_state('kafka.ca.keystore.saved')
+
+
+@when('endpoint.certificates.departed')
+def clear_certificates():
+    ca_path = caPath()
+    ca_keystore = caKeystore()
+    ca_keystore_secret = keystoreSecret()
+    if os.path.exists(ca_path):
+        os.remove(ca_path)
+    if os.path.exists(ca_keystore):
+        os.remove(ca_keystore)
+    if os.path.exists(ca_keystore_secret):
+        os.remove(ca_keystore_secret)
+
+    certs_paths = unitdata.kv().get('layer.tls-client.cert-paths', {})
+    for cert_type in ('server', 'client'):
+        crt_path = crtPath(cert_type)
+        key_path = keyPath(cert_type)
+        keystore_path = keystore(cert_type)
+        if os.path.exists(crt_path):
+            os.remove(crt_path)
+        if os.path.exists(key_path):
+            os.remove(key_path)
+        if os.path.exists(keystore_path):
+            os.remove(keystore_path)
+        remove_state('kafka.{}.keystore.saved'.format(cert_type))
+        data_changed('kafka_{}_certificate'.format(cert_type), {})
+
+        for common_name, paths in certs_paths.get(cert_type, {}).items():
+            data_changed('layer.tls-client.'
+                         '{}.{}'.format(cert_type, common_name), {})
+
+    # We need to clean up data because the underlying layer
+    # failes to clear flags and data_changed states when
+    # we remove {charm}-easyrsa relation.
+    unitdata.kv().unsetrange(
```

Comment:
Consider something like this for readability, ease of adding additional keys if we find more that we need to add, etc:
```
cleanup_unitdata_keys = (
  'reactive.tls_client',
  'reactive.data_changed.endpoint.certificates',
  ...
)
for k in cleanup_unitdata_keys:
  unitdata.kv().unsetrange(None, k)
```

Reply:
done